src/client/components/dashboard/SettingsModal.tsx

```diff
@@ -0,0 +1,412 @@
+import _ from 'lodash'
+import React, { FC, useEffect, useState } from 'react'
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  VStack,
+  FormControl,
+  FormLabel,
+  Switch,
+  HStack,
+  Spacer,
+  Divider,
+  Link,
+  Input,
+  IconButton,
+  Button,
+  Table,
+  Tbody,
+  Tr,
+  Td,
+  useDisclosure,
+} from '@chakra-ui/react'
+import { Box, Text } from '@chakra-ui/layout'
+import { BiLinkExternal, BiTrash } from 'react-icons/bi'
+import { useMutation, useQuery, useQueryClient } from 'react-query'
+import { useHistory, useLocation, useParams } from 'react-router-dom'
+import { useAuth } from '../../contexts'
+import { useStyledToast } from '../common/StyledToast'
+import { ConfirmDialog } from '../ConfirmDialog'
+import { DefaultTooltip } from '../common/DefaultTooltip'
+import { getApiErrorMessage } from '../../api'
+import { CheckerService } from '../../services'
+
+import { Checker } from '../../../types/checker'
+import { EmbedField } from '../common/EmbedField'
+import { CollaboratorUser } from '../../../types/user'
+
+type CollaboratorsTableProps = {
+  collaboratorsData: CollaboratorUser[] | undefined
+  onDelete: (collaboratorEmail: string) => () => void
+}
+
+type LocationState = {
+  checker: Checker
+}
+
+const CollaboratorsList: FC<CollaboratorsTableProps> = ({
+  collaboratorsData,
+  onDelete,
+}) => {
+  const { user } = useAuth()
+  const tableItems = collaboratorsData?.map((collaborator) => {
+    return (
+      <Tr key={collaborator.id}>
+        <Td px={0} py={0}>
+          {collaborator.email +
+            (collaborator.email === user?.email ? ' (you)' : '')}
+        </Td>
+        <Td px={0} py={0}>
+          {collaborator.UserToChecker.isOwner ? (
+            <Text color="grey">Owner</Text>
+          ) : (
+            <Text>Editor</Text>
+          )}
+        </Td>
+        <Td px={0} py={0}>
+          {collaborator.UserToChecker.isOwner ? (
+            <Box height="40px" />
+          ) : (
+            <IconButton
```

Comment:
I think the delete option should be disabled for the logged in user. Otherwise, it is a bit strange to delete himself as a collaborator.

![image](https://user-images.githubusercontent.com/3666479/126948773-6b292a79-1679-4eab-928a-9d3c94fa048f.png)